text/0000-ember-new-lang.md

```diff
@@ -0,0 +1,239 @@
+
+
+
+#### Invalid Language Codes
+
+Language codes are verified against [is-language-code](https://www.npmjs.com/package/is-language-code). (see [examples of valid ISO country codes](https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes))
+
+If an invalid language code is given such as `--lang en-UK` the indended output should be a shell error that will halt the build.
+
+```bash
+ember new my-app --lang en-UK
+Unrecognized language subtag, "uk".
+```
+
+#### No argument passed
+Additionally, if the `--lang` flag is used but no argument is defined, e.g., `ember new my-app --lang`, the build will also be halted. An error message and usage info should be shown in these cases:
+
+```bash
+# Detect trailing option (declared)
+ember new my-app --lang --skip-git
+
+An error with the `--lang` flag returned the following message:
+  Detected lang specification starting with command flag `-`.
+  Is `--skip-git` meant to be an ember-cli command option?
+  This issue is likely caused by using the `--lang` flag without a specification.
+Information about using the `--lang` flag:
+  The `--lang` flag sets the base human language of the app in index.html
+  If used, the lang option must specfify a valid language code.
+  For default behavior, remove the flag.
+  See `ember <command> help` for more information.
+```
+
+```bash
+# Detect trailing option (undeclared)
+ember new my-app --lang
+
+An error with the `--lang` flag returned the following message:
+  Detected lang specification starting with command flag `-`.
+  Is `--disable-analytics` meant to be an ember-cli command option?
+  This issue is likely caused by using the `--lang` flag without a specification.
+Information about using the `--lang` flag:
+  The `--lang` flag sets the base human language of the app in index.html
+  If used, the lang option must specfify a valid language code.
+  For default behavior, remove the flag.
+  See `ember <command> help` for more information.
+```
+
+#### Common Misunderstandings: Programming Languages
+
+A developer may encounter the flag and make incorrect assumptions about what it can mean. Such as, `-l typescript` or `-l glimmer`. Such incorrect assumptions will be manually caught by the implementation and the developer will be shown a friendly error message such as the following:
+
+
+```bash
+ember new my-app --lang=typescript
+
+An error with the `--lang` flag returned the following message:
+  Trying to set the app programming language to `typescript?`
+  This is not the intended usage of the `--lang` flag.
+Information about using the `--lang` flag:
+  The `--lang` flag sets the base human language of the app in index.html
+  If used, the lang option must specfify a valid language code.
+  For default behavior, remove the flag.
+  See `ember <command> help` for more information.
+```
+
+
+
+## How we teach this
```

Comment:
Great work on this RFC! I enjoy how well reasoned it is and its focus.

For that reason, I would like to see the "How we teach this" section expanded with some of the knowledge in the motivation section. The "Testing it out" was particularly illumating to me as a content creator in understanding the implications of the `lang` attribute.

We have a [Language attribute](https://guides.emberjs.com/release/accessibility/application-considerations/#toc_language-attribute) section in the Ember Guides that could also be updated to mention the new flag?

Reply:
@locks yes, that seems correct. I’ll work on it. Thanks for the feedback!